runtime/src/main/java/org/corfudb/runtime/view/workflows/AddNode.java

```diff
@@ -55,4 +62,11 @@ protected boolean verifyRequest(@NonNull Layout layout) {
     public String toString() {
         return this.getClass().getSimpleName() + " " + nodeForWorkflow;
     }
+
+    @Override
+    protected Optional<ManagementClient> getOrchestrator(){
+
+        return Optional.of(runtime.getLayoutView()
```

Comment:
You are right. Actually, in AddNode, we would have to ping it more than once. Since the node pings itself, there could be cases when after the restart and immediate add, the node's routers are not ready for a couple of milliseconds, so some pings won't be able to go through.